### PR TITLE
refactor done seprated constants from config

### PR DESCRIPTION
**Pull request by LijuJoseJJ**
this PR fixes  https://github.com/EYBlockchain/nightfall_3_private/issues/158

### CODE REVIEW

**Review by Westlad — CHANGES_REQUESTED** [review]
I think we need to be careful about the definition of a `config` item and a `constant`.  A `constant` will never depend on the details of our deployment.  An example is `ZERO: 0x000....`.  A `config` item is something that is deployment dependent, for example the name of a Mongo collection or the height of a Merkle tree.  There are only a few grey areas, such as the definition of calldata types.  These could arguably be either but are best categorised as constants because they will actually never be dependent on the details of a deployment.

**Comment by LijuJoseJJ** [comment]
> I think we need to be careful about the definition of a `config` item and a `constant`. A `constant` will never depend on the details of our deployment. An example is `ZERO: 0x000....`. A `config` item is something that is deployment dependent, for example the name of a Mongo collection or the height of a Merkle tree. There are only a few grey areas, such as the definition of calldata types. These could arguably be either but are best categorised as constants because they will actually never be dependent on the details of a deployment.

changes accommodated

File: common-files/constants/constants.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  "EXCLUDE_DIRS": "common",
+  "PROOF_QUEUE": "generate-proof",
+  "MAX_QUEUE": 5,
+  "ZKP_KEY_LENGTH": 32,
+  "CONFIRMATION_POLL_TIME": 1000,
+  "CONFIRMATIONS": 12,
+  "NODE_HASHLENGTH": 32,
+  "DEFAULT_ACCOUNT_NUM": 10,
+  "ZERO": "0x0000000000000000000000000000000000000000000000000000000000000000",
+  "HASH_TYPE": "poseidon",
```
Comment (Westlad) [inline common-files/constants/constants.json:17]: Note a constant - we change the hash type from time to time

File: common-files/constants/constants.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  "TXHASH_TREE_HASH_TYPE": "keccak256",
```
Comment (Westlad) [inline common-files/constants/constants.json:18]: Not a constant

File: common-files/constants/constants.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
+  "CONTRACT_ARTIFACTS": "/app/build/contracts",
+  "PROPOSERS_CONTRACT_NAME": "Proposers",
+  "SHIELD_CONTRACT_NAME": "Shield",
+  "CHALLENGES_CONTRACT_NAME": "Challenges",
+  "STATE_CONTRACT_NAME": "State",
+  "BLOCK_PROPOSED_EVENT_NAME": "BlockProposed",
+  "EXCLUDE_DIRS": "common",
+  "PROOF_QUEUE": "generate-proof",
+  "MAX_QUEUE": 5,
```
Comment (Westlad) [inline common-files/constants/constants.json:10]: Not a constant

File: common-files/constants/constants.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  "ZKP_KEY_LENGTH": 32,
+  "CONFIRMATION_POLL_TIME": 1000,
```
Comment (Westlad) [inline common-files/constants/constants.json:12]: Not a constant

File: common-files/constants/constants.json [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  "CONFIRMATIONS": 12,
```
Comment (Westlad) [inline common-files/constants/constants.json:13]: Not a constant

File: common-files/constants/constants.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
+{
+  "CONTRACT_ARTIFACTS": "/app/build/contracts",
```
Comment (Westlad) [inline common-files/constants/constants.json:2]: not a constant

File: common-files/constants/constants.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  "PROPOSERS_CONTRACT_NAME": "Proposers",
+  "SHIELD_CONTRACT_NAME": "Shield",
+  "CHALLENGES_CONTRACT_NAME": "Challenges",
+  "STATE_CONTRACT_NAME": "State",
+  "BLOCK_PROPOSED_EVENT_NAME": "BlockProposed",
+  "EXCLUDE_DIRS": "common",
```
Comment (Westlad) [inline common-files/constants/constants.json:8]: not a constant

File: common-files/constants/constants.json [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  "PROOF_QUEUE": "generate-proof",
```
Comment (Westlad) [inline common-files/constants/constants.json:9]: not a constant

File: common-files/constants/constants.json [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  "MAX_QUEUE": 5,
+  "ZKP_KEY_LENGTH": 32,
+  "CONFIRMATION_POLL_TIME": 1000,
+  "CONFIRMATIONS": 12,
+  "NODE_HASHLENGTH": 32,
+  "DEFAULT_ACCOUNT_NUM": 10,
```
Comment (Westlad) [inline common-files/constants/constants.json:15]: Not a constant

File: common-files/constants/constants.json (hunk 8 above)
Comment (LijuJoseJJ) [reply common-files/constants/constants.json:9]: moved to config

File: common-files/constants/constants.json (hunk 7 above)
Comment (LijuJoseJJ) [reply common-files/constants/constants.json:8]: moved to config

File: common-files/constants/constants.json (hunk 6 above)
Comment (LijuJoseJJ) [reply common-files/constants/constants.json:2]: moved to config

File: common-files/constants/constants.json (hunk 5 above)
Comment (LijuJoseJJ) [reply common-files/constants/constants.json:13]: moved to config

File: common-files/constants/constants.json (hunk 4 above)
Comment (LijuJoseJJ) [reply common-files/constants/constants.json:12]: moved to config

File: common-files/constants/constants.json (hunk 3 above)
Comment (LijuJoseJJ) [reply common-files/constants/constants.json:10]: moved to config

**Review by Westlad — APPROVED** [review]

File: common-files/constants/constants.json (hunk 9 above)
Comment (LijuJoseJJ) [reply common-files/constants/constants.json]: moved to config

File: common-files/constants/constants.json (hunk 2 above)
Comment (LijuJoseJJ) [reply common-files/constants/constants.json]: moved to config

File: common-files/constants/constants.json (hunk 1 above)
Comment (LijuJoseJJ) [reply common-files/constants/constants.json]: moved to config

**Review by druiz0992 — APPROVED** [review]

**Pull request merged**